LPC55xxx secure/non-secure split

agansari opened the pull request:
Adds NS definitions for CPU0 and CPU1.
Preparing platform for future development, dual core, mpu, etc.

Fixes #16330

agansari committed:
dts: nxp_lpc55S6x_ns.dtsi added

Added non-secure mappings for LPC55S6x SoC family.

Signed-off-by: Andrei Gansari <andrei.gansari@nxp.com>

agansari committed:
dts: lpc55xxx add mpu

MPU added to LPC55xxx dts configuration.

Signed-off-by: Andrei Gansari <andrei.gansari@nxp.com>

agansari committed:
boards: lpcxpresso55s69_cpu1 definitions

LPC55S69 CPU1 definition added.
Dual Core is not enabled!
Definitions related to dual core split of SoC's CPUs.

Signed-off-by: Andrei Gansari <andrei.gansari@nxp.com>

agansari committed:
soc: lpc55xxx mpu areas defined from dts

Number of MPU regions obtained from MPU.

Signed-off-by: Andrei Gansari <andrei.gansari@nxp.com>

agansari committed:
dts: lpc55s69 non-secure all flash regions

Added reserved partition and bootrom partiotions.

Signed-off-by: Andrei Gansari <andrei.gansari@nxp.com>

agansari committed:
boards: lpcxpresso55s69_cpu0 partitioning

Moved code-partition to SRAMX and Flash0 split into multiple partitions.

Signed-off-by: Andrei Gansari <andrei.gansari@nxp.com>

agansari committed:
boards: add target lpcxpresso55s69_ns

Adds NS (Non-Secure) target for LPC55S69 CPU0.

Signed-off-by: Andrei Gansari <andrei.gansari@nxp.com>

agansari committed:
soc: lpc55xxx define secure/non-secure core

Define secure related features on the lpc55xxx soc cores.

Signed-off-by: Andrei Gansari <andrei.gansari@nxp.com>

d3zd3z commented on boards/arm/lpcxpresso55s69/lpcxpresso55s69_cpu0.dts:
```diff
@@ -64,14 +65,49 @@
+		};
+		scratch_partition: partition@10060000 {
+			label = "image-scratch";
+			reg = <0x10060000 0x0003D800>;
```
I believe the DT convention is to use lowercase for hex constants.

d3zd3z commented on dts/arm/nxp/nxp_lpc55S6x_ns.dtsi:
```diff
@@ -0,0 +1,158 @@
+				write-block-size = <512>;
+			};
+
+			flash_reserved:flash@9D800 {
```
This hex value, and the one below should be lowercase.

d3zd3z commented on dts/arm/nxp/nxp_lpc55S6x_ns.dtsi:
```diff
@@ -0,0 +1,158 @@
+
+		usart0:usart@40086000 {
+			compatible = "nxp,lpc-usart";
+			reg = <0x40086000 0xE44>;
```
Likewise. Lowercase.

d3zd3z reviewed (changes requested):
Other than a minor issue with the case of hex values, this looks good.

microbuilder commented on boards/arm/lpcxpresso55s69/lpcxpresso55s69_cpu1.yaml:
```diff
@@ -0,0 +1,18 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+
+identifier: lpcxpresso55s69_cpu0
```
I may have misunderstood something, but should this be `lpcxpresso55s69_cpu1` if the NS app is running on the second core?

microbuilder commented on boards/arm/lpcxpresso55s69/lpcxpresso55s69_ns.yaml:
```diff
@@ -0,0 +1,20 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+
+identifier: lpcxpresso55s69_cpu0
```
As above, should this be `lpcxpresso55s69_cpu1` if the NS app is running on the second core?

agansari replied:
@microbuilder lpcxpresso55s69_ns refers to CPU0 built with NS address ranges.
CPU1 is NS only, guess I should document this.

agansari committed:
boards: lpcxpresso55s69 available targets description

Describes the targets LPC55S69 support, CPU0 secure and non-secure, CPU1
non-secure.

Signed-off-by: Andrei Gansari <andrei.gansari@nxp.com>

agansari commented:
@microbuilder added documentation about the possible targets

d3zd3z commented on dts/arm/nxp/nxp_lpc55S6x_ns.dtsi:
```diff
@@ -0,0 +1,158 @@
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			flash0:flash@0{
```
There should be a space after the colon (here and several below).

d3zd3z reviewed (changes requested):
One minor whitespace issue, but otherwise looks good.
